### PR TITLE
Refactoring random to ParcelsRandom class

### PR DESCRIPTION
**Pull request by erikvansebille**
This implements #873. Note that this renaming of the random function is not backward-compatibel; users can choose two ways to adapt their code

1. Change 
    ```diff
    - from parcels import random
    + from parcels import ParcelsRandom as random
    ```

2. Change 
    ```diff
    - from parcels import random
    + from parcels import ParcelsRandom
    ``` 
    and rename all calls to `random` functions to `ParcelsRandom`

### CODE REVIEW

**Review by daanreijnders — APPROVED** [review]
Looks all good to me 👍

**Comment by CKehl** [comment]
On my list - scheduled for Thursday morning

File: parcels/codegenerator.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def visit_Name(self, node):
             node = StatusCodeNode(math, ccode='')
         elif node.id == 'math':
             node = MathNode(math, ccode='')
-        elif node.id == 'random':
+        elif node.id == 'ParcelsRandom':
```
Comment (CKehl) [inline parcels/codegenerator.py:258]: interesting that this change was required - I can't remember that I needed to do that. Well spotted!

File: parcels/rng.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 __all__ = ['seed', 'random', 'uniform', 'randint', 'normalvariate', 'expovariate', 'vonmisesvariate']
 
 
-class Random(object):
+class RandomC(object):
```
Comment (CKehl) [inline parcels/rng.py:16]: like the name - encapsulates the reason of why that is needed

File: tests/test_kernel_language.py [hunk 3]
```diff
@@ -265,7 +265,7 @@ def kernel(particle, fieldset, time):
 
 
 def random_series(npart, rngfunc, rngargs, mode):
-    random = parcels_random if mode == 'jit' else py_random
+    random = ParcelsRandom if mode == 'jit' else py_random
     random.seed(1234)
     func = getattr(random, rngfunc)
     series = [func(*rngargs) for _ in range(npart)]
```
Comment (CKehl) [inline tests/test_kernel_language.py:271]: yes, exactly - these parts are the ones that become quite difficult of all the different libraries are named 'random'.

**Review by CKehl — APPROVED** [review]
Yes - the changes look all very good. I think this now makes it more robust and reasy to switch RNG libraries and avoid scipy-jit deviations.

**Pull request merged**